### PR TITLE
Make Metis datatypes consistent with libmesh's

### PR DESCRIPTION
**Pull request by jwpeterson**
This branch ensures that the sizes of METIS's `idx_t` and `real_t` are consistent with libmesh's `dof_id_type` and `Real`, respectively.  It also applies some patches from [PETSc's parmetis distro](http://ftp.mcs.anl.gov/pub/petsc/externalpackages/parmetis-4.0.3-p1.tar.gz) pointed out by @jedbrown.

### CODE REVIEW

**Comment by jwpeterson** [comment]
I've also tested this against a PETSc configured with
```
--download-parmetis=1
--download-metis=1
--with-64-bit-indices
```

and libmesh configured with:
```
--with-metis=PETSc
--with-dof-id-bytes=8
```

and it seems to work.

File: include/partitioning/parmetis_partitioner.h [hunk 1]
```diff
@@ -111,23 +114,10 @@ class ParmetisPartitioner : public Partitioner
   vectormap<dof_id_type, dof_id_type> _global_index_by_pid_map;
 
   /**
-   * Data structures used by ParMETIS to describe the connectivity graph
-   * of the mesh.  Consult the ParMETIS documentation.
+   * Pointer to the Parmetis-specific data structures.  Lets us avoid
+   * including parmetis.h here.
    */
```
Comment (roystgnr) [inline include/partitioning/parmetis_partitioner.h]: Appreciate this

**Comment by roystgnr** [comment]
I reserve the right to scream when this causes some strange breakage for me.

But as far as I can tell from the diffs this should do nothing but avoid strange breakage.  :+1: to merge when you're ready.

**Comment by jwpeterson** [comment]
> I reserve the right to scream when this causes some strange breakage for me.

OK, thanks for reviewing.

**Pull request merged**